### ISSUE TITLE
A combination of union, transformer and default infers the wrong type

### ISSUE THREAD

**Issue by ohana54** [colinhacks/zod#332]
Hi,
I'm using v3.0.0-alpha.4 and encountered a case where adding a default causes the inferred type to be wrong.

Example:

```typescript
const schema = z.array(
  z.union([
    z.string().transform(name => ({ name })),
    z.object({ name: z.string() })
  ])
);

const schemaWithDefault = schema.default([]);

type Schema = z.infer<typeof schema>;
/* 
infers the correct type:
({
    name: string;
} | {
    name: string;
})[]
*/

type SchemaWithDefault = z.infer<typeof schemaWithDefault>;
/* 
infers the wrong type:
(string | {
    name: string;
})[]
*/
```

Any idea what's going on here?
Thanks!

**Comment by colinhacks** [comment]
This is fixed in the newest alpha `zod@3.0.0-alpha.7`. Thanks for pointing this out!

**Comment by ohana54** [comment]
Amazing, thanks, I tried and it worked great!

A side issue - the new version adds an npm dependency on `eslint-plugin-unused-imports`, should these be dev dependencies? Should I open a new issue for it?
